tests/qnn/test_keras.py

```diff
@@ -101,28 +142,26 @@ def test_weight_shape_unspecified(self, get_circuit, output_dim):
             KerasLayer(c, w, output_dim)
 
     @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
-    def test_var_pos(self, get_circuit, monkeypatch, output_dim):
+    def test_var_pos(self):
         """Test if a TypeError is raised when instantiated with a variable number of positional
         arguments"""
-        c, w = get_circuit
-
-        class FuncPatch:
-            """Patch for variable number of keyword arguments"""
+        dev = qml.device("default.qubit", wires=1)
+        weight_shapes = {"w1": (3, 3), "w2": 1}
 
-            sig = c.func.sig
-            var_pos = True
-            var_keyword = False
-
-        with monkeypatch.context() as m:
-            m.setattr(c, "func", FuncPatch)
+        @qml.qnode(dev, interface="tf")
+        def circuit(inputs, w1, w2, *args):
+            return qml.expval(qml.PauliZ(0))
 
-            with pytest.raises(TypeError, match="Cannot have a variable number of positional"):
-                KerasLayer(c, w, output_dim)
+        with pytest.raises(TypeError, match="Cannot have a variable number of positional"):
+            KerasLayer(circuit, weight_shapes, output_dim=1)
 
     @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
     def test_var_keyword(self, get_circuit, monkeypatch, output_dim):
         """Test if a TypeError is raised when instantiated with a variable number of keyword
         arguments"""
+        if qml.tape_mode_active():
+            pytest.skip("This functionality is supported in tape mode.")
```

Comment:
Just wanted to confirm this is as intended, i.e., no typos? (just looks a bit unusual :laughing:)